cmd/kubeadm/app/util/apiclient/idempotency.go

```diff
@@ -293,3 +293,24 @@ func PatchNode(client clientset.Interface, nodeName string, patchFn func(*v1.Nod
 	// then the retries end and the error is returned.
 	return wait.Poll(constants.APICallRetryInterval, constants.PatchNodeTimeout, PatchNodeOnce(client, nodeName, patchFn))
 }
+
+// GetConfigMapWithRetry tries to retrieve a ConfigMap using the given client,
+// retrying if isAllowedError returns true for the received error during the get
+// action of the ConfigMap.
+func GetConfigMapWithRetry(client clientset.Interface, namespace, name string) (*v1.ConfigMap, error) {
+	var cm *v1.ConfigMap
+	var lastError error
+	err := wait.ExponentialBackoff(clientsetretry.DefaultBackoff, func() (bool, error) {
```

Comment:
Do we have a specific error condition that occurs on the race of new members joining?  
/cc @gyuho

Reply:
we are not convinced this issue is directly related to etcd member join.

Reply:
At minimum we need a //TODO + open an issue to fix in the next releases.

Reply:
PR updated to include the TODO comment.

Reply:
TODO added and ticket created: https://github.com/kubernetes/kubeadm/issues/1606